tmt/steps/prepare/reboot.py

```diff
@@ -0,0 +1,103 @@
+import dataclasses
+import os
+from typing import Any, List, Optional, cast
+
+import click
+import pkg_resources
+
+import tmt
+import tmt.steps.prepare
+import tmt.utils
+from tmt.steps.common import RebootCommon
+from tmt.steps.execute import TMT_REBOOT_SCRIPT
+from tmt.steps.provision import Guest
+
+# Scripts source directory
+SCRIPTS_SRC_DIR = pkg_resources.resource_filename(
+    'tmt', 'steps/execute/scripts')
+
+SCRIPTS = (
+    TMT_REBOOT_SCRIPT,
+    )
+
+# TODO: remove `ignore` with follow-imports enablement
+
+
+@dataclasses.dataclass
+class PrepareRebootData(tmt.steps.prepare.PrepareStepData):
+    script: List[str] = dataclasses.field(default_factory=list)
+
+    _normalize_script = tmt.utils.NormalizeKeysMixin._normalize_string_list
+
+
+@tmt.steps.provides_method('reboot')
+class PrepareReboot(
+        tmt.steps.prepare.PreparePlugin,
+        RebootCommon
+        ):
+    """
+    Reboot system via provided script
+    Example config:
+    prepare:
+        how: reboot
+        script: ./reboot-script
+    """
+
+    _data_class = PrepareRebootData  # type: ignore[assignment]
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> Any:
+        """ Prepare command line options """
+        return cast(List[tmt.options.ClickOptionDecoratorType], [
+            click.option(
+                '-n', '--name', metavar='NAME',
+                help='Set name of the reboot step.'),
+            click.option(
+                '-n', '--script', metavar='NAME',
+                help='Set path to the reboot script.')
+            ]) + super().options(how)
+
+    def default(self, option: str, default: Optional[Any] = None) -> Any:
+        """ Return default data for given option """
+        if option == 'name':
+            return "Reboot script name"
+        if option == 'reboot':
```

Comment:
This option does not seem to be defined, there's only `how`, `name` and `script`.

Reply:
Yep, it was actually typo (there should have been different option -> script). Dropped anyway.